reflection/src/main/java/net/kyori/examination/reflection/ReflectiveExaminablePropertiesImpl.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of examination, licensed under the MIT License.
+ *
+ * Copyright (c) 2018-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.examination.reflection;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import net.kyori.examination.ExaminableProperty;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+final class ReflectiveExaminablePropertiesImpl implements ReflectiveExaminableProperties {
+  private final List<Supplier<ExaminableProperty>> properties;
+
+  private ReflectiveExaminablePropertiesImpl(final List<Supplier<ExaminableProperty>> properties) {
+    this.properties = properties;
+  }
+
+  public static ReflectiveExaminablePropertiesImpl forFields(final Object object) {
+    final List<Supplier<ExaminableProperty>> properties = new ArrayList<>();
+    for(final Field field : object.getClass().getDeclaredFields()) {
+      final Examine examine = field.getAnnotation(Examine.class);
+      if(examine != null) {
+        field.setAccessible(true); // TODO(kashike): does this break on new Java versions?
```

Comment:
i believe this sorta works -- modular applications that want to use examination would have to open their modules to deep reflection from the reflective properties provider module.